### DISCUSSION TITLE
What is the target audience of a firejail profile?

### DISCUSSION THREAD

**Discussion by crocket** [netblue30/firejail#4972] [Q&A]
A tech expert who knows how to selectively allow some interpreters required by an arbitrary editor launched by a sandboxed program?
Someone who understands ${RUNUSER} and other concepts intimately?

A beginner who just wants to improve security and privacy without hassle?

**Comment by SkewedZeppelin** [comment]
Back when I was writing profiles my goal was keep them sane enough to not break
your mum's workflow, but also strict enough that someone wanting security
wouldn't be concerned that a door was left wide open.

**Comment by ghost** [comment]
> A beginner who just wants to improve security and privacy without hassle?

Back when @SkewedZeppelin was writing profiles, I personally was uneasily starting out with Firejail, having read somewhere it was a must-have thing if you were interested in privacy and security of your Linux desktop. Looking back I sure felt/was very much like that beginner you describe. Eager to sandbox all the applications I used on a daily basis, once and for all, Install And Forget, starting with the web browser - the One True Gateway to The Internets. Seeing what has become of that single tool one realises pretty soon that these privacy/security issues everyone is going on about are in fact of mythical proportions. Learn by example, browse through the repo, take notes, pose questions, file issues. Or not. Time will pass regardless and chances are slim you'll be any closer to the Goal.

I honestly don't think there exists such a concept as 'target audience' in the Firejail community. And maybe that's a Good Thing. Or at least not necessarily a Bad Thing. Explicitly written out rulesets, that, when meticulously followed step by step, lead to guaranteed sandboxing succes. Perhaps the [templates](https://github.com/netblue30/firejail/tree/master/etc/templates) come close. But those too assume some level of familiarity with Firejail's _and_ the sandboxed applications. Again, with hindsight, the one, singular most important lesson that stands out is that without hassle pretty much nothing truly moves anywhere, ever. Fast-forwarding to the present, and here we are again, still beginners, maybe just maybe only slightly less hassle-free-minded.

As applications evolve and hunger to integrate as much features as possible, the more difficult it becomes to make the kind of informed decision @SkewedZeppelin hints at. That's the real challenge IMO, not the amount of knowledge one has to accumulate to write profiles. Mirroring this ongoing race I/O complexity (Singularity anyone?), Firejail itself, by default, also evolves along. That makes life very difficult for us beginners. Changing options, newly introduced options, combinable options, more features, more issues, more Food for (Git-)Historians. This is not a conspiracy of arrogant, highly-skilled and sophisticated individuals. Very little 'stuff' exists in a vacuum, technically or socially speaking. And as noble as the goal of hitting that sweet-spot between sane-simple-secure might be, better prepare for uphill battles.

Then again, armies of privacy/security AI bots might roll in tomorrow and turn this repo upside-down and inside-out. Firejail v 1.0 - the final solution - come and get it, its covering all potentially installable software as of right now! Consisting of exactly one, super-tiny, none-SUID executable and a handful of basic sandboxing rulesets. Individual application profiles? Gone. Science-Fiction? Absolutely. Impossible? Very likely. Improbable? Not really. Something like the current `--build` option, yet `Artificially Enhanced`, churning out `Certified Strictest Sandboxes` at dark-hole speeds. We can all lay back and enjoy cool drinks in the Firejail Treehouse.

**Comment by crocket** [comment]
In particular, I'm considering whether I should support certain editors like neovim and kakoune in a profile for a program that launches editors.
